util/helm/client.go

```diff
@@ -102,29 +110,25 @@ func fileExist(filePath string) (bool, error) {
 	return true, nil
 }
 
-func (c *nativeHelmChart) ensureHelmChartRepoPath() error {
-	c.repoLock.Lock(c.repoPath)
-	defer c.repoLock.Unlock(c.repoPath)
-
-	err := os.Mkdir(c.repoPath, 0700)
-	if err != nil && !os.IsExist(err) {
+func (c *nativeHelmChart) CleanChartCache(chart string, version string) error {
+	cachePath, err := c.getCachedChartPath(chart, version)
+	if err != nil {
 		return err
 	}
-	return nil
+	return os.RemoveAll(cachePath)
 }
 
-func (c *nativeHelmChart) CleanChartCache(chart string, version string) error {
-	return os.RemoveAll(c.getCachedChartPath(chart, version))
+func createTempDir() (string, error) {
+	tempDir := path.Join(os.TempDir(), uuid.New().String())
```

Comment:
Should we use `uuid.NewRandom()` and handle errors? I imagine errors from that function are rare, but would be good to avoid any unnecessary panics.

Reply:
makes sense.